config/prow/jobs/config.yaml

```diff
@@ -3428,6 +3428,70 @@ presubmits:
       - name: test-account
         secret:
           secretName: test-account
+  - name: pull-knative-test-infra-check-config
+    agent: kubernetes
+    context: pull-knative-test-infra-check-config
+    always_run: true
+    rerun_command: "/test pull-knative-test-infra-check-config"
+    trigger: "(?m)^/test (all|pull-knative-test-infra-check-config),?(\\s+|$)"
+    decorate: true
+    path_alias: knative.dev/test-infra
+    branches:
+    - "legacy"
```

Comment:
I am a bit confused, if all these jobs have "legacy" as branches, do they get run then. Why are there so many jobs marked as legacy? Is this part of the generator that takes in the prow config for the legacy branches and automatically generates all this? Like I feel there may be two changes here that is not made very clear by the commit message.

1) We seem to have gotten rid of release-0.10 and replaced it with this fake "legacy" branch. 
2) There is an extra periodic job that does the config check
3) How do I know if these changes affect the production cluster vs. staging cluster?

I don't know what the effects of 1) is and I am not sure how 2) was checked previously (manually call make?). For 3) it seems we only have a single prow config file generated right, so what is the part about production vs. staging clusters?

Reply:
1) Either `release-0.10` or `legacy` is added for making it work for config generator, the Prow jobs themselves do not have any effect as there is only one master branch in this repo.
(This is actually a bug in config generator, but we haven't got time to fix it)

2) We don't have an extra periodic job that does the config check. Previously we are doing this check in the presubmit-build-tests, but it was not clear and missing some important checks. This PR is moving the check to a separate Prow job.

3) We have a set of config files for both production and staging Prow, see `config/prow` and `config/prow-staging`. The only configuration files they share are the cluster configs in `config/prow/cluster` (which in the future we will also make a copy in `config/prow-staging` while we implement Prow auto-update)